keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -0,0 +1,1215 @@
+  // AllPorts cannot be "false" when it is set 
+  // AllPorts allows the user to select all ports for all protocols, thus not 
+  // selecting traffic based on L4 principles.
+  // If "true" then all ports are selected for the all protocols
+  AllPorts   *bool
+
+  // List of ports for outgoing traffic.
+  // Each item in this list is combined using a logical OR. If this field is
+  // empty or missing, this rule matches no ports (traffic not restricted by port).
+  // If this field is present and contains at least one item, then this rule 
+  // allows/denies/passes traffic only if the traffic matches at least one port in the list.
+  // +optional
+  Ports      []AdminNetworkPolicyPort 
+}
+
+
+// AdminNetworkPolicyPort describes a port to select
+type AdminNetworkPolicyPort struct {
+  // The protocol (TCP, UDP, or SCTP) which traffic must match. If not specified, this
+  // field defaults to TCP.
+  // +optional
+  Protocol   *v1.Protocol
+
+  // The port on the given protocol. This can either be a numerical or named
+  // port on a pod. If this field is not provided, this matches no port names and
+  // numbers.
+  // If present, only traffic on the specified protocol AND port will be matched.
+  // +optional
+  Port       *intstr.IntOrString
+    
+  // If set, indicates that the range of ports from port to endPort, inclusive,
+  // should be allowed by the policy. This field cannot be defined if the port field
+  // is not defined or if the port field is defined as a named (string) port.
+  // The endPort must be equal or greater than port.
+  // This feature is in Beta state and is enabled by default.
+  // It can be disabled using the Feature Gate "NetworkPolicyEndPort".
+  // +optional
+  EndPort     *int32 
+}
+
+const (
+  // RuleActionPass enables admins to provide exceptions to ClusterNetworkPolicies and delegate this rule to
+  // K8s NetworkPolicies.
+  AdminNetPolRuleActionPass    AdminNetPolRuleAction = "Pass"
+  // RuleActionDeny enables admins to deny specific traffic.
+  AdminNetPolRuleActionDeny    AdminNetPolRuleAction = "Deny"
+  // RuleActionAllow enables admins to specifically allow certain traffic.
+  AdminNetPolRuleActionAllow   AdminNetPolRuleAction = "Allow"
+)
+
+// AdminNetworkPolicyPeer defines an in-cluster peer to allow traffic to/from. 
+// Exactly one of the selector pointers should be set for a given peer. 
+type AdminNetworkPolicyPeer struct {
+  Namespaces       *NamespaceSet
+  NamespacedPods   *NamespaceAndPodSet
+}
+
+// NamespaceSet defines a flexible way to select Namespaces in a cluster.
+// Exactly one of the selectors should be set.  If a consumer observes none of 
+// its fields are set, they should assume an option they are not aware of has 
+// been specified and fail closed.
+type NamespaceSet struct {
+  // Self cannot be "false" when it is set.  
+  // If Self is "true" then all pods in the subject's namespace are selected.
+  Self                *bool
```

Comment:
Does it make sense to get rid of Self, NotSelf and replace them with a single field `Invert`?